envoy/http/filter_factory.h

```diff
@@ -113,9 +113,10 @@ class FilterChainFactory {
    *    returns false if this upgrade type is not configured, and no filter chain is created.
    */
   using UpgradeMap = std::map<std::string, bool>;
-  virtual bool createUpgradeFilterChain(absl::string_view upgrade,
-                                        const UpgradeMap* per_route_upgrade_map,
-                                        FilterChainManager& manager) const PURE;
+  virtual bool createUpgradeFilterChain(
+      absl::string_view upgrade, const UpgradeMap* per_route_upgrade_map,
+      FilterChainManager& manager,
+      const Http::FilterChainOptions& option = EmptyFilterChainOptions{}) const PURE;
```

Comment:
Nit: The code isalready in the HTTP namespace, no need for `Http::` infront of FilterChainOptions.

Might also make sense to call this "options" to match the call above and since the object itself is plural (FilterChainOptions)

Reply:
If you think it will make sense, I can create a separate PR to fix this inconsistency.

Reply:
yea that works for me.

Reply:
As promised, I created a PR with fixes https://github.com/envoyproxy/envoy/pull/35500